src/nm.c

```diff
@@ -430,7 +430,12 @@ write_dot1x_auth_parameters(const NetplanAuthenticationSettings* auth, GKeyFile
         g_key_file_set_string(kf, "802-1x", "identity", auth->identity);
     if (auth->anonymous_identity)
         g_key_file_set_string(kf, "802-1x", "anonymous-identity", auth->anonymous_identity);
-    if (auth->password && auth->key_management != NETPLAN_AUTH_KEY_MANAGEMENT_WPA_PSK)
+    if (auth->password
+        && ((   auth->key_management != NETPLAN_AUTH_KEY_MANAGEMENT_WPA_PSK
+             && auth->key_management != NETPLAN_AUTH_KEY_MANAGEMENT_WPA_SAE
+            )
+            || auth->eap_method != NETPLAN_AUTH_EAP_NONE)
+    )
```

Comment:
note (non-blocking): I guess this should work, but maybe a comment should be added to describe what's going on. Also, I wonder if it could be simplified when always storing the PSK (if available) by something like `auth->password && !auth->psk`?

Reply:
The condition `auth->password && !auth->psk?` wouldn't work because we could have both passwords at the same time.

Reply:
OK. I see you changed it to make use of the `is_auth_key_management_psk()` helper and added a comment. Thanks! That's easier to grasp.